lib/mqtt.c

```diff
@@ -692,6 +694,16 @@ static CURLcode mqtt_do(struct Curl_easy *data, bool *done)
   return CURLE_OK;
 }
 
+static CURLcode mqtt_done(struct Curl_easy *data,
+                          CURLcode status, bool premature)
+{
+  struct MQTT *mq = data->req.p.mqtt;
+  (void)status;
+  (void)premature;
+  Curl_safefree(mq->sendleftovers);
+  return CURLE_OK;
+}
+
```

Comment:
While I haven't done any testing, from reading this seems perfectly reasonable. Shouldn't however `mqtt_doing` set `mq->sendleftovers` to `NULL` as well as a belts-and-suspenders type operation?

Reply:
Re-reading it again I concur.

Reply:
thanks for the check!